integration-tests/cli/product-json-to-xlsx.it.js

```diff
@@ -0,0 +1,885 @@
+            expect(product[1]).toEqual(expect.objectContaining(slug))
+          })
+
+          it('contains `searchKeywords`', () => {
+            const searchKeywords = { 'searchKeywords.en': 'Multi Tool;Swiss | Army | Knife' }
+            expect(product[0]).toEqual(
+              expect.objectContaining(searchKeywords)
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining(searchKeywords)
+            )
+          })
+
+          it('contains resolved namedPath of `categories`', () => {
+            const categories = 'Parent Category>child Category'
+            expect(product[0]).toEqual(expect.objectContaining({ categories }))
+            expect(product[1]).toEqual(expect.objectContaining({ categories }))
+          })
+
+          it('contains resolved `state`', () => {
+            const state = 'stateKey'
+            expect(product[0]).toEqual(expect.objectContaining({ state }))
+            expect(product[1]).toEqual(expect.objectContaining({ state }))
+          })
+
+          it('contains resolved `taxCategory`', () => {
+            const tax = 'new-tax-category'
+            expect(product[0]).toEqual(expect.objectContaining({ tax }))
+            expect(product[1]).toEqual(expect.objectContaining({ tax }))
+          })
+
+          it('contains variants `SKUs`', () => {
+            expect(product[0]).toEqual(
+              expect.objectContaining({ sku: 'M00F56YSS' })
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining({ sku: 'M00F56YSS-11' })
+            )
+          })
+
+          it('contains variants `keys`', () => {
+            expect(product[0]).toEqual(
+              expect.objectContaining({ variantKey: 'master-var-111' })
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining({ variantKey: 'normal-var-222' })
+            )
+          })
+
+          // Test for custom attributes
+          it('contains variants custom attributes', () => {
+            expect(product[0]).toEqual(
+              expect.objectContaining({
+                'another-text-attribute': 'Another Master Var attr',
+              })
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining({
+                'another-text-attribute': 'Another First Var attr',
+              })
+            )
+          })
+        })
+      })
+
+      describe('WITH HEADERS:: write products to `XLSX` file', () => {
+        let xlsxFile
+        let products = []
+        const templateFile = `${__dirname}/helpers/product-headers.csv`
+
+        beforeAll(async done => {
+          xlsxFile = tmp.fileSync({ postfix: '.xlsx' }).name
+          const [stdout, stderr] = await exec(
+            `${exporter} -p ${projectKey} -s | ${binPath} -p ${projectKey} -t ${templateFile} --referenceCategoryBy name -o ${xlsxFile}`
+          )
+          expect(stdout).toBeTruthy()
```

Comment:
wondering why this pattern with the beforeAll here too

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper